### DISCUSSION TITLE
how to listen for a event only in specific rooms not globally. In the docs I couldn't found a way to do that!

### DISCUSSION THREAD

**Discussion by arnav-kr** [socketio/socket.io#3951] [Q&A] [answered]
I want something like this

I have an admin room in my app which only admins can access and I want to listen to client events only in that room

I want something like this
```
io.in("room1").on(event,callback);
```

**Comment by darrachequesne** [comment, marked as answer]
Hi! I see two possible solutions:

- you can check if the socket is actually in the restricted room

```js
io.on("connection", (socket) => {
  socket.on("my-event", () => {
    const isAllowed = socket.rooms.has("admin");
    if (isAllowed) {
      // ...
    }
  });
});
```

- or use a [namespace](https://socket.io/docs/v4/namespaces/) (that's their exact purpose)


```js
io.of("/admin").on("connection", (socket) => {
  socket.on("my-event", () => {
    // ...
  });
});
```

**Reply by arnav-kr** [reply]
let me try it

**Reply by arnav-kr** [reply]
> let me try it

Yes It worked Thanks For the Help
